lib/dgram.js

```diff
@@ -573,6 +715,25 @@ Socket.prototype.address = function() {
   return out;
 };
 
+Socket.prototype.remoteAddress = function() {
+  healthCheck(this);
+
+  const state = this[kStateSymbol];
+  if (state.connectState !== CONNECT_STATE_CONNECTED) {
+    // eslint-disable-next-line no-restricted-syntax
```

Comment:
For all of the places where you do this, could you define (or reuse an existing) custom error?